Core/include/Acts/EventData/TrackProxyConcept.hpp

```diff
@@ -0,0 +1,30 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/EventData/MultiTrajectoryBackendConcept.hpp"
+#include "Acts/EventData/TrackContainerBackendConcept.hpp"
+#include "Acts/EventData/Types.hpp"
+
+#include <concepts>
+
+namespace Acts {
+
+template <typename T>
+concept TrackProxyConcept = requires() {
```

Comment:
At this time, I think this seems reasonable. Ultimately, it might make sense to check that it's an instance of the `TrackProxy` template class. I've experimented with this for something else but didn't manage to do it yet. Anyway, this is not critical for this PR.